### PR TITLE
Refactors body markings into bodypart overlays

### PR DESCRIPTION
**Pull request by Time-Green**
## About The Pull Request
Refactors body markings (lizard and moths) into the bodypart overlays system
I've also renamed the icon states for lizard bodymarkings (downstreams with lots of lizard body markings might have problems with this)

## Why It's Good For The Game
Gives us better controle over /human appearances. I didn't add any means of actually changing them on the person (aside from genetics. how would that even work? with a waterproof marker?), but makes it easy for when/if I do. Also unloads mutant_bodyparts more, moving is yet closer to full species dedatumization

## Changelog
:cl:
refactor: Lizard and moth markings now use the bodypart overlay system
/:cl:

### CODE REVIEW

File: code/__DEFINES/mobs.dm [hunk 1]
```diff
@@ -797,6 +797,12 @@ GLOBAL_LIST_INIT(layers_to_offset, list(
 /// This organ allows restyling with enamel restyling (like a fucking file or something?). It's for horns and shit
 #define EXTERNAL_RESTYLE_ENAMEL (1 << 3)
 
+// Bitflags to indicate which body markings we're allowed to use on the species
```
Comment (MrMelbert) [inline code/__DEFINES/mobs.dm:800]: "Bitflags"

These defines are not necessary anyways, just put the typepath in where it's used

File: code/modules/mining/lavaland/megafauna_loot.dm [hunk 2]
```diff
@@ -749,7 +749,7 @@
 				"wings" = "None",
 				"frills" = "None",
 				"spines" = "Long",
-				"body_markings" = "Dark Tiger Body",
+				"lizard_markings" = "Dark Tiger Body",
```
Comment (MrMelbert) [inline code/modules/mining/lavaland/megafauna_loot.dm:752]: You don't have to do it in this pr but we really should go around and replace this kinda stuff with
`/datum/sprite_accessory/lizard_markings/dtiger::name`

Comment (Time-Green) [reply code/modules/mining/lavaland/megafauna_loot.dm:752]: that's be a really good idea, all of this code is filled with hardcoded strings like this

Comment (tralezab) [reply code/modules/mining/lavaland/megafauna_loot.dm:752]: that is SUCH a good idea lets go

**Comment by MrMelbert** [comment]
I'm glad I made that test

File: code/__DEFINES/mobs.dm (hunk 1 above)
Comment (Time-Green) [reply code/__DEFINES/mobs.dm:800]: yeah idk why I even did that

File: code/modules/client/preferences/_preference.dm [hunk 3]
```diff
@@ -107,6 +107,10 @@ GLOBAL_LIST_INIT(preference_entries_by_key, init_preference_entries_by_key())
 	/// will show the feature as selectable.
 	var/relevant_mutant_bodypart = null
 
+	/// If the selected species has this in its /datum/species/body_markings,
+	/// will show the feature as selectable.
+	var/relevant_body_markings = null
```
Comment (AnturK) [inline code/modules/client/preferences/_preference.dm:112]: These all should probably be unified into less copypasty preference constraints in another PR

File: code/modules/mob/living/carbon/human/_species.dm [hunk 4]
```diff
@@ -2110,3 +2080,48 @@ GLOBAL_LIST_EMPTY(features_by_species)
 			return fixed_mut_color
 
 	return null
+
+/// Add species appropriate body markings
+/datum/species/proc/add_body_markings(mob/living/carbon/human/hooman)
+	for(var/datum/bodypart_overlay/simple/body_marking/markings as anything in body_markings) //loop through possible species markings
+		markings = new markings() // made to die... mostly because we cant use initial on lists but its convenient and organized
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/_species.dm:2087]: please just put this in separate variable, this is not a hot path for it to matter

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/client/preferences/_preference.dm (hunk 3 above)
Comment (Time-Green) [reply code/modules/client/preferences/_preference.dm:112]: if all goes well, sooner rather than later

**Pull request merged**